docs/envoy.md

```diff
@@ -0,0 +1,326 @@
+# Life of an OSM Request
+
+## A bit on Envoy
+
+OSM heavily uses envoy for proxying requests, applying mTLS, and exporting statistics. Envoy itself has 5 main
+configuration mechanisms for request manipulation:
+
+* ListenerConfigs
+* RouteConfigs
+* EndpointConfigs
+* ClusterConfigs
+* SecretConfigs
+
+### Listeners
+
+Envoy itself intercepts all requests by reconfiguring the IP tables on the Pod to reroute incoming and outbound traffic
```

Comment:
I see what you are saying here but I am not sure if the sentence is correct. Envoy does not configure the IP tables, OSM does that through `initcontainer`.